proposals/0000-scan-takewhile-dropwhile.md

```diff
@@ -0,0 +1,138 @@
+# Feature name
+
+* Proposal: TBD
+* Author(s): [Kevin Ballard](https://github.com/kballard)
+* Status: **Awaiting review**
+* Review manager: TBD
+
+## Introduction
+
+Add 3 new `SequenceType` functions `scan(_:combine:)`, `takeWhile(_:)`, and
+`dropWhile(_:)`, with overrides as appropriate on `CollectionType`,
+`LazySequenceType`, and `LazyCollectionType`, as well as a global function
+`iterate(_:apply:)`.
+
+Swift-evolution thread:
+[Proposal: Add scan, takeWhile, dropWhile, and iterate to the stdlib](https://lists.swift.org/pipermail/swift-evolution/Week-of-Mon-20151228/004690.html)
+
+## Motivation
+
+The Swift standard library provides many useful sequence manipulators like
+`dropFirst(_:)`, `filter(_:)`, etc. but it's missing a few common methods that
+are quite useful.
+
+## Proposed solution
+
+Add the following extension to `SequenceType`:
+
+```swift
+extension SequenceType {
+  /// Returns an array containing the results of
+  ///
+  ///     p.reduce(initial, combine: combine)
+  ///
+  /// for each prefix `p` of `self`, in order from shortest to longest.
```

Comment:
It would be good to say explicitly whether the empty prefix is included, and maybe even to be explicit that the longest prefix is `self` itself.